include/deal.II/base/symmetric_tensor.h

```diff
@@ -2916,7 +2916,9 @@ operator * (const SymmetricTensor<rank,dim,Number> &t,
   // (as well as with switched arguments and double<->float).
   typedef typename ProductType<Number,OtherNumber>::type product_type;
   SymmetricTensor<rank,dim,product_type> tt(t);
-  tt *= product_type(factor);
+  product_type new_factor;
+  new_factor = factor;
```

Comment:
This is not obvious -- it works because, apparently, you have a copy operator but not a conversion constructor.

Can you add a comment in front of these two lines to explain the issue and give an example (in the comment) of types for which this works but didn't before?